modules/nft/src/lib.rs

```diff
@@ -0,0 +1,229 @@
+				orml_nft::Module::<T>::create_class(&who, metadata, data)?;
+
+				Self::deposit_event(RawEvent::CreatedClass(who, next_id));
+				Ok(())
+			})?;
+		}
+
+		#[weight = 10_000]
+		pub fn mint(origin, to: T::AccountId, class_id: <T as orml_nft::Trait>::ClassId, metadata: CID, quantity: u32) {
+			with_transaction_result(|| {
+				let who = ensure_signed(origin)?;
+				let class_info = orml_nft::Module::<T>::classes(class_id).ok_or(Error::<T>::ClassIdNotFound)?;
+				ensure!(who == class_info.owner, Error::<T>::NoPermission);
+				let deposit = T::CreateTokenDeposit::get();
+				let owner: T::AccountId = T::ModuleId::get().into_sub_account(class_id);
+				let total_deposit = deposit * (quantity as u128);
+				<T as Trait>::Currency::transfer(&who, &owner, total_deposit)?;
+				<T as Trait>::Currency::reserve(&owner, total_deposit)?;
+
+				let data = TokenData { deposit };
+				let data = T::ConvertTokenData::from(data);
+				let data = Into::<<T as orml_nft::Trait>::TokenData>::into(data);
+				for _ in 0..quantity {
+					orml_nft::Module::<T>::mint(&to, class_id, metadata.clone(), data.clone())?;
+				}
+
+				Self::deposit_event(RawEvent::MintedToken(who, to, class_id, quantity));
+				Ok(())
+			})?;
+		}
+
+		#[weight = 10_000]
+		pub fn transfer(origin, to: T::AccountId, token: (<T as orml_nft::Trait>::ClassId, <T as orml_nft::Trait>::TokenId)) {
+			with_transaction_result(|| {
+				let who = ensure_signed(origin)?;
+				let class_info = orml_nft::Module::<T>::classes(token.0).ok_or(Error::<T>::ClassIdNotFound)?;
+				let data: T::ConvertClassData = From::from(class_info.data);
+				ensure!(data.into().properties.contains(&ClassProperty::Transferable), Error::<T>::NonTransferable);
+
+				let token_info = orml_nft::Module::<T>::tokens(token.0, token.1).ok_or(Error::<T>::TokenIdNotFound)?;
+				ensure!(who == token_info.owner, Error::<T>::NoPermission);
+
+				orml_nft::Module::<T>::transfer(&who, &to, token)?;
+
+				Self::deposit_event(RawEvent::TransferedToken(who, to, token.0, token.1));
+				Ok(())
+			})?;
+		}
+
+		#[weight = 10_000]
+		pub fn burn(origin, token: (<T as orml_nft::Trait>::ClassId, <T as orml_nft::Trait>::TokenId)) {
+			with_transaction_result(|| {
+				let who = ensure_signed(origin)?;
+				let class_info = orml_nft::Module::<T>::classes(token.0).ok_or(Error::<T>::ClassIdNotFound)?;
+				let data: T::ConvertClassData = From::from(class_info.data);
+				ensure!(data.into().properties.contains(&ClassProperty::Burnable), Error::<T>::NonBurnable);
+
+				let token_info = orml_nft::Module::<T>::tokens(token.0, token.1).ok_or(Error::<T>::TokenIdNotFound)?;
+				ensure!(who == token_info.owner, Error::<T>::NoPermission);
+
+				orml_nft::Module::<T>::burn(&who, token)?;
+				let owner: T::AccountId = T::ModuleId::get().into_sub_account(token.0);
+				let data: T::ConvertTokenData = From::from(token_info.data);
+				let data: TokenData = data.into();
+				<T as Trait>::Currency::unreserve(&owner, data.deposit);
```

Comment:
use `repatriate_reserved` to unreserve and transfer at same time

Reply:
https://github.com/paritytech/substrate/blob/master/frame/balances/src/lib.rs#L1263
If `to` account is not exist, `repatriate_reserved` will return `DeadAccount`.
But `transfer` did not do this check.

Reply:
ok. you can add that to comment so next reviewer won't ask the same question.